packages/strapi-plugin-i18n/admin/src/hooks/useDefaultLocales/index.js

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useReducer, useRef } from 'react';
+import { request } from 'strapi-helper-plugin';
+import reducer, { initialState } from './reducer';
+
+const fetchDefaultLocalesList = async (dispatch, signal) => {
```

Comment:
Would you like to try to use `react-query` instead? You can take the work you did in the #9246 ?

Reply:
yes sure, we can try this 👍🏻